core/drivers/crypto/se050/core/rsa.c

```diff
@@ -463,13 +463,21 @@ static TEE_Result do_encrypt(struct drvcrypt_rsa_ed *rsa_data)
 {
 	switch (rsa_data->rsa_id) {
 	case DRVCRYPT_RSA_PKCS_V1_5:
+		return encrypt_es(TEE_ALG_RSAES_PKCS1_V1_5,
```

Comment:
Why is `rsa_data->hash_algo` not `TEE_ALG_RSAES_PKCS1_V1_5` in this case? It looks weird that you need to throw in a constant here while the OAEP case is dynamic.

Reply:
yes. I do agree, that looked weird to me too...I think the hash_algo field in the interface should have been just named algo and the algorithm always set.... but that is the way the API was written so I kind of assumed it was intentional.

maybe @clementfaure can comment?

Reply:
@jforissier so to be clear, the patch under review just obeys the API. 

if there is a problem, it is with the Crypto API itself so its maintainers should comment.

Reply:
Ok then for "crypto: drivers: se050: fix rsa encrypt/decrypt":

`Acked-by: Jerome Forissier <jerome@forissier.org>`